src/app/feature/consultant/services/consultant-store/consultant-store.service.ts

```diff
@@ -69,13 +69,58 @@ export class ConsultantStore implements OnDestroy {
   /**
    * Build a new consultant object from existing object with updated properties.
    * @param data A partial Consultant object containing the data to update
+   * @param index optional arg which says which index in edu or cert array to replace
    */
-  private updatedConsultantFactory(data: Partial<Consultant>): Consultant {
+  private updatedConsultantFactory(data, index?: number): Consultant {
     // Copy current consultant to preserve immutability
-    const consultantCopy = {
-      ...this.consultant,
-      ...data
-    };
+    var consultantCopy: Consultant;
```

Comment:
changing what i pass in through data might also get rid of this ugly if else tree. for now its needed to persist/update the education or certifications arrays that are on the Consultant object.